cmd/status-backend/server/server.go

```diff
@@ -53,12 +53,28 @@ func (s *Server) signalHandler(data []byte) {
 	s.lock.Lock()
 	defer s.lock.Unlock()
 
+	var disconnected []*websocket.Conn
+
 	for connection := range s.connections {
-		err := connection.WriteMessage(websocket.TextMessage, data)
+		err := connection.SetWriteDeadline(time.Now().Add(5 * time.Second))
+		if err != nil {
+			log.Error("failed to set write deadline", "error", err)
+			disconnected = append(disconnected, connection)
```

Comment:
It's safe to remove the key rightaway in the loop, no need for temporary array.

https://go.dev/play/p/kdVQBURZE4B